tests/cpp11_condition_variable/Makefile

```
@@ -6,7 +6,7 @@ CXXEXFLAGS += -std=c++11
 
 # nucleo-f303k8 doesn't have enough RAM to run the test so we reduce the stack
 # size for every thread
-ifneq (,$(filter nucleo-f303k8,$(BOARD)))
+ifneq (,$(filter nucleo-f303k8 nucleo-f334r8,$(BOARD)))
   CFLAGS += -DTHREAD_STACKSIZE_DEFAULT=512
 endif
```

IMO it is better not to do this based on boards. E.g. this wouldn't hurt on other Cortex-M boards, as it is unlikely that those do have different stack consumption. But building with the same stack size increases reproducibility and chances that an MPU stack guard detects overflows.

---

Not sure I understand the end of the comment, you would rather change the stack size based on `$(CPU_CORE)`?

```suggestion
ifneq (,$(filter cortex-m0%,$(CPU_CORE)))
  CFLAGS += -DTHREAD_STACKSIZE_DEFAULT=512
endif
```

---

My point is that stack requirements depend on the program and the platform it is run on, not on the available RAM of a specific board. This `Makefile` might mislead users to assume that there is indeed some value in reducing the stack size for "small" boards - but actually one should instead change the applications to use the smallest (plus some wiggle room) thread size possible.

---

How about adding

```
CFLAGS += -DTHREAD_STACKSIZE_DEFAULT=THREAD_STACKSIZE_SMALL
```

? (Hoping that the small stack size is not defined as `THREAD_STACKSIZE_DEFAULT / 4` or so :-D)

---

Why the heck does the C++ thread class not to specify the thread size, and just blindly uses `THREAD_STACKSIZE_MAIN`? The cleanest IMO would be to modify `main.cpp` to use `THREAD_STACKSIZE_SMALL` as stack size instead.

---

Technically that would require more testing, which is the only reason why I would not want to do that.

---

This would also not be really related to this PR, as the issues is preexisting. I might also look into extending the C++ thread API to allow specifying the thread size and use this test as motivating use case to bundle with that.

---

So keep at is for this one?